src/Symfony/Component/Mailer/Tests/EventListener/DkimSignedMessageListenerTest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Tests\EventListener;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Mailer\Envelope;
+use Symfony\Component\Mailer\Event\MessageEvent;
+use Symfony\Component\Mailer\EventListener\MessageListener;
+use Symfony\Component\Mailer\EventListener\DkimSignedMessageListener;
+use Symfony\Component\Mailer\Exception as MailerException;
+use Symfony\Component\Mime\Exception as MimeException;
+use Symfony\Component\Mime\Address;
+use Symfony\Component\Mime\Header\Headers;
+use Symfony\Component\Mime\Header\MailboxListHeader;
+use Symfony\Component\Mime\Header\UnstructuredHeader;
+use Symfony\Component\Mime\Message;
+
+class DkimSignedMessageListenerTest extends TestCase
+{
+    public function testMessageEqualsWhenNoSigner()
+    {
+        $listener = new DkimSignedMessageListener();
+        $message = new Message();
+        $envelope = new Envelope(new Address('sender@example.com'), [new Address('r1@example.com')]);
+        $event = new MessageEvent($message, $envelope, 'default');
+
+        $listener->onMessage($event);
+
+        $this->assertEquals($message, $event->getMessage());
+    }
+
+    public function testDkimMessageErrorWithoutDomain()
+    {
+        $listener = new DkimSignedMessageListener('dkim', 'file:///path/to/private-key.key');
+        $message = new Message();
+        $envelope = new Envelope(new Address('sender@example.com'), [new Address('r1@example.com')]);
+        $event = new MessageEvent($message, $envelope, 'default');
+
+        $this->expectException(MailerException\InvalidArgumentException::class);
+        
+        $listener->onMessage($event);
+    }
+
+    public function testSignedDkimMessage()
+    {
+        $listener = new DkimSignedMessageListener('dkim', 'file:///path/to/private-key.key', 'www.symfony.com', 'sfe');
+        $message = new Message();
+        $envelope = new Envelope(new Address('sender@example.com'), [new Address('r1@example.com')]);
+        $event = new MessageEvent($message, $envelope, 'default');
+
+        $this->expectException(MimeException\InvalidArgumentException::class); // An `Unable to load private key exception` is expected here
```

Comment:
Instead of the comment, you may additionally use `expectExceptionMessage()` to assert this 🙂 Same for `SmimeSignedMessageListenerTest`

Reply:
Thanks Alexandre. Great feedback.